### DISCUSSION TITLE
How to add support for functions calls with chains?

### DISCUSSION THREAD

**Discussion by urjitbhatia** [tmc/langchaingo#329] [Q&A]
Hi,

I am trying to use the `ConversationalRetrievalQA` chain with `StuffDocuments` and `CondenseQuestion` chains and I'd love to be able to get function call recommendations from openAI. Currently, this isn't easy since chains options don't pass on function call params to the llm calls. Any ideas about how to proceed with this?

For now, I copied the chain into my project and reimplemented the guts a bit. Are other people running into similar roadblocks?

**Comment by devalexandre** [comment]
> Hi,
> 
> I am trying to use the `ConversationalRetrievalQA` chain with `StuffDocuments` and `CondenseQuestion` chains and I'd love to be able to get function call recommendations from openAI. Currently, this isn't easy since chains options don't pass on function call params to the llm calls. Any ideas about how to proceed with this?
> 
> For now, I copied the chain into my project and reimplemented the guts a bit. Are other people running into similar roadblocks?

have a  sample code?
